src/services/swap/providers/lightning-labs/loop-service.ts

```
@@ -0,0 +1,172 @@
+      const fee = resp.getSwapFeeSat()
+      if (fee) return true
+    } catch (error) {
+      console.log(error)
+    }
+    return false
+  }
+
+  const swapOut = async function (
+    amount,
+    maxSwapFee?,
+  ): Promise<SwapOutResult | SwapServiceError> {
+    const fee = maxSwapFee ? maxSwapFee : 20000
+    try {
+      const request = new LoopOutRequest()
+      request.setAmt(amount)
+      request.setMaxSwapFee(fee)
+      request.setMaxPrepayRoutingFee(fee)
+      request.setMaxSwapFee(fee)
+      request.setMaxPrepayAmt(fee)
+      request.setMaxMinerFee(fee)
+      request.setSweepConfTarget(2)
+      request.setHtlcConfirmations(1)
+      request.setSwapPublicationDeadline(600) // TODO - play with these params --fast
+      const resp = await clientSwapOut(request)
+      const swapOutResult: SwapOutResult = {
+        htlcAddress: resp.getHtlcAddress(),
+        serverMessage: resp.getServerMessage(),
+        swapId: resp.getId(),
+        swapIdBytes: resp.getIdBytes().toString(),
+      }
+      return swapOutResult
+    } catch (error) {
+      return new SwapServiceError(error)
+    }
+  }
+
+  const swapListener = function (): SwapClientReadableStream<SwapListenerResponse> {
+    try {
+      const request = new MonitorRequest()
+      const listener = swapClient.monitor(request)
+      // listener.on("data" | "status" | "end")
+      listener.on("data", (data: SwapStatus & SwapStatusResultWrapper) => {
+        listener.pause()
+        // parse data to our interface
+        const stateVal = data.getState()
+        let state
+        try {
+          state = Object.keys(SwapState).find((key) => SwapState[key] === stateVal)
+        } catch (e) {
+          state = SwapState.FAILED
+        }
+        let message
+        try {
+          const failureReason = data.getFailureReason()
+          message = Object.keys(FailureReason).find(
+            (key) => FailureReason[key] === failureReason,
+          )
+        } catch (e) {}
```

we should typically not silence errors